doc/whats_new.rst

```diff
@@ -62,6 +62,16 @@ Model Selection Enhancements and API Changes
     the corresponding parameter is not applicable. Additionally a list of all
     the parameter dicts are stored at ``results_['params']``.
 
+  - **Renaming of `n_folds` and `n_iter` to `n_splits`**
+
+    The number of folds (i.e. the number of partitions of a dataset) is not 
```

Comment:
I think this doesn't need so much motivation here, you just need to say something like "In the new `model_selection` module, some parameter names have changed: the `n_folds` parameter in x, y, z is now `n_splits`; the `n_iter` parameter in a, b, c is now `n_splits`". Please use `:class:` as appropriate.